src/trigrams.rs

```diff
@@ -1,6 +1,6 @@
 use std::collections::HashMap;
 
-pub fn get_trigrams_with_positions(text : &String) -> HashMap<String, u32> {
+pub fn get_trigrams_with_positions(text : &str) -> HashMap<String, u32> {
```

Comment:
The reason it works is deref-coercion. String derefs to str, so where a function takes `&str`, you can pass `&String`. This also works with `&[T]`/`&Vec<T>`.

In general, functions should always take `&str`/`&[T]` as these are the more general types.